### DISCUSSION TITLE
Feature Request: API Route Search Functionality in API Documentation

### DISCUSSION THREAD

**Discussion by rolandverner** [dedoc/scramble#619] [Ideas]
I would like to request the addition of a search functionality for API routes within the API documentation. This feature would significantly improve the usability and efficiency of navigating through the documentation, particularly for developers who work with numerous endpoints and need to locate specific routes quickly.

### Feature Requirements:

1. Search Box for Routes: A search box at the top of the API documentation that allows users to type keywords or parts of route names to filter the available routes.

2. Autocomplete Suggestions: As the user types in the search box, provide autocomplete suggestions based on the available routes for quick selection.

3. Filter by HTTP Method: An option to filter the search results by HTTP method (e.g., GET, POST, PUT, DELETE) to refine results further.

4. Highlight Search Results: Once a route is selected from the search results, highlight or scroll directly to the corresponding documentation section for easy access.

5. Clear Search Button: A button to clear the search and return to the full API documentation view without needing to refresh the page.

### Benefits:

- Enhanced developer productivity by reducing the time spent searching through documentation.
- Improved user experience, especially for teams managing multiple APIs or working with complex API structures.
- Streamlined navigation, allowing users to find necessary information with fewer clicks and scrolls.

### Conclusion:

Implementing a search feature for API routes within the documentation would add tremendous value to developers and streamline the development process. I believe this enhancement will foster more efficient use of the API and support a better documentation experience.

Thank you for considering this request. I look forward to any updates regarding this feature.

**Comment by romalytvynenko** [comment]
Hey @rolandverner 

I'm working on own UI that will has this feature. 

You can check the upcoming aesthetic here: https://playground.dedoc.co/p/3ad87fd4-fe43-47e3-ab6a-2dab6846985c#operations/job.store

**Reply by laadtushar** [reply]
how is this coming along?
